addon/services/log.js

```diff
@@ -0,0 +1,348 @@
+/**
```

Comment:
I think we need to ask all developers, whos branches are using user-settings service or log service, to merge `develop` branch into their branches after merge this pull request.